pennylane/spin/lattice.py

```diff
@@ -220,136 +217,72 @@ def add_edge(self, edge_indices):
-    )
-    return lattice_chain
-
-
-def _square(n_cells, boundary_condition=False, neighbour_order=1):
-    r"""Generates a square lattice"""
-    vectors = [[1, 0], [0, 1]]
-    positions = [[0, 0]]
-    n_cells = n_cells[0:2]
-    lattice_square = Lattice(
-        n_cells=n_cells,
-        vectors=vectors,
-        positions=positions,
-        neighbour_order=neighbour_order,
-        boundary_condition=boundary_condition,
-    )
-
-    return lattice_square
-
-
-def _rectangle(n_cells, boundary_condition=False, neighbour_order=1):
-    r"""Generates a rectangle lattice"""
-    vectors = [[1, 0], [0, 1]]
-    positions = [[0, 0]]
-
-    n_cells = n_cells[0:2]
-    lattice_rec = Lattice(
-        n_cells=n_cells,
-        vectors=vectors,
-        positions=positions,
-        neighbour_order=neighbour_order,
-        boundary_condition=boundary_condition,
-    )
-
-    return lattice_rec
-
-
-def _honeycomb(n_cells, boundary_condition=False, neighbour_order=1):
-    r"""Generates a honeycomb lattice"""
-    vectors = [[1, 0], [0.5, math.sqrt(3) / 2]]
-    positions = [[0, 0], [0.5, 0.5 / 3**0.5]]
-
-    n_cells = n_cells[0:2]
-    lattice_honeycomb = Lattice(
-        n_cells=n_cells,
-        vectors=vectors,
-        positions=positions,
-        neighbour_order=neighbour_order,
-        boundary_condition=boundary_condition,
-    )
-
-    return lattice_honeycomb
-
-
-def _triangle(n_cells, boundary_condition=False, neighbour_order=1):
-    r"""Generates a triangular lattice"""
-    vectors = [[1, 0], [0.5, math.sqrt(3) / 2]]
-    positions = [[0, 0]]
-
-    n_cells = n_cells[0:2]
-    lattice_triangle = Lattice(
-        n_cells=n_cells,
-        vectors=vectors,
-        positions=positions,
-        neighbour_order=neighbour_order,
-        boundary_condition=boundary_condition,
-    )
-
-    return lattice_triangle
-
-
-def _kagome(n_cells, boundary_condition=False, neighbour_order=1):
-    r"""Generates a kagome lattice"""
-    vectors = [[1, 0], [0.5, math.sqrt(3) / 2]]
-    positions = [[0.0, 0], [-0.25, math.sqrt(3) / 4], [0.25, math.sqrt(3) / 4]]
-
-    n_cells = n_cells[0:2]
-    lattice_kagome = Lattice(
-        n_cells=n_cells,
-        vectors=vectors,
-        positions=positions,
-        neighbour_order=neighbour_order,
-        boundary_condition=boundary_condition,
-    )
-
-    return lattice_kagome
-
-
-# TODO Check the efficiency of this function with a dictionary instead.
 def _generate_lattice(lattice, n_cells, boundary_condition=False, neighbour_order=1):
     r"""Generates the lattice object for a given shape and n_cells.
 
     Args:
-        lattice (str): Shape of the lattice. Input Values can be ``'chain'``, ``'square'``, ``'rectangle'``, ``'honeycomb'``, ``'triangle'``, or ``'kagome'``.
+        lattice (str): Shape of the lattice. Input Values can be ``'chain'``, ``'square'``, ``'rectangle'``,
+              ``'honeycomb'``, ``'triangle'``, ``'kagome'``, ``'lieb'``, ``'cubic'``, ``'bcc'``, or ``'fcc'``.
```

Comment:
Where do we give more information regarding these lattice structures in the docs?

Reply:
I don't understand, do you mean we should add wiki links/papers to these?

Reply:
We can have a separate PR to improve the docs overall and also expose Lattice to users. @ddhawan11 Please keep note of such recommendation for that future PR.